site/_posts/2017-10-24-api-contributors-guide.md

```diff
@@ -0,0 +1,252 @@
+    def order_resource(type, id, data)
+      coffee = resource_search(id, type, collection_class(type))
+      coffee.order(data)
+    rescue => err
+      raise BadRequestError, "Coffee not ordered - #{err}"
+    end 
+  end 
+end 
+```
+The above example demonstrates a couple of the basic methods and formats that you'll see used throughout the API.
+
+* `collection_class(type)`
+    * Based off of the `type` of request (in this case, `:coffee`), it will return the `klass` defined in the `api.yml` (`Coffee`)
+* `resource_search(id, type, class)`
+    * Resource search will return the resource corresponding to the ID that was requested, or return a `NotFoundError` if it was not found
+    * Resource search is used as opposed to a simple `Coffee.find(id)` because the results are filtered through access control, ensuring nobody is working with a resource they don't have permissions for
+* data
+    * The data input is the request body that was passed in with the order request. I like my coffee `{"cream": false, "sugar": false}`, so that is the request body that would be passed to my order.
+
+## Subcollections
+Oftentimes, we need subcollections on a resource to return child resources. Coffee is great by itself, but it's even better with donuts. And it's your lucky day - every coffee has a set of donuts picked to pair perfectly with it. Sounds like the perfect job for a subcollection. 
+
+Although the Donuts could also be a collection (they can still be enjoyed on their own), this example will only outline the subcollection actions for reading and deleting them. 
+```
+    :donuts:
+        :description: Donuts
+        :options: 
+        - :subcollection
+        :verbs: *gp
+        :klass: Donut
+        :subcollection_actions:
+            :get: 
+            - :name: read
+              :identifier: donut_show_list 
+            :post:
+            - :name: delete
+              :identifier: donut_delete
+        :subresource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: delete
+              :identifier: coffee_delete
+```
+The above configuration is very similar to that of a collection, only it defines `:subcollection_actions:` and `:subresource_actions:`. The coffee options will also need to be updated to include the Donuts subcollection:
+```
+    :coffee:
+        :description: Coffee
+        :options: 
+        - :collection
+        :verbs: *gpd
+        :klass: Coffee
+        :subcollections:
+        - :donuts
+```
+
+Subcollection actions are defined in [modules](https://github.com/ManageIQ/manageiq-api/tree/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api/subcollections) and included in the collection controller they are a subcollection of. 
+
+Below is a simple example of what the `Donuts` module would look like:
+```
+module Api
+  module Subcollections
+    module Donuts
+      def donuts_query_resource(object)
```

Comment:
Similarly here I think a little about the naming scheme and the ✨ behind it would be great